lib/grammers-client/src/client/auth.rs

```diff
@@ -261,6 +265,58 @@ impl Client {
             Err(InvocationError::Rpc(RpcError { name, .. })) if name.starts_with("PHONE_CODE_") => {
                 Err(SignInError::InvalidCode)
             }
+            Err(InvocationError::Rpc(RpcError { name, .. }))
+                if name.starts_with("SESSION_PASSWORD_NEEDED") =>
+            {
+                Err(SignInError::PasswordRequired)
+            }
+            Err(error) => Err(SignInError::Other(error)),
+        }
+    }
+
+    pub async fn two_factor_auth(
+        &mut self,
+        password: Vec<u8>,
+    ) -> Result<tl::types::User, SignInError> {
+        let request = tl::functions::account::GetPassword {};
```

Comment:
We might want to perform this call when we get `SESSION_PASSWORD_NEEDED`, so that the `SignInError::PasswordRequired` can include the hint (that the caller can display somehow) along with a "token" (similar to how send code request gives you one for sign in) that has to be used to call this method.